server/cast/varchar.go

```diff
@@ -146,6 +146,20 @@ func varcharExplicit() {
 			return d, nil
 		},
 	})
+	framework.MustAddExplicitTypeCast(framework.TypeCast{
+		FromType: pgtypes.VarChar,
+		ToType:   pgtypes.Oid,
+		Function: func(ctx framework.Context, val any, targetType pgtypes.DoltgresType) (any, error) {
+			out, err := strconv.ParseInt(strings.TrimSpace(val.(string)), 10, 64)
```

Comment:
With the integer-to-oid stuff earlier allowing for negative values, I also threw this into Postgres:
```
SELECT ('-1'::text)::oid;
```
Which worked without surprise. I also try to add tests for limits as well (for things that have limits like integers), verifying that the limit passes, but limit+1 fails. Doing that, I ended up getting to:
```
SELECT ('-4294967295'::text)::oid;
```
Which also passes. I'm not sure how valid this is in other casts or explicit/implicit/assignment contexts, but it works for `text` so we should check for the other string types too.